src/utils.js

```diff
@@ -275,8 +282,10 @@ ko.utils = (function () {
                     while (current !== last) {
                         continuousNodeArray.push(current);
                         current = current.nextSibling;
-                        if (!current) // Won't happen, except if the developer has manually removed some DOM elements (then we're in an undefined scenario)
-                            return;
+                        if (!current) { // Won't happen, except if the developer has manually removed some DOM elements (then we're in an undefined scenario)
```

Comment:
With the change above (new rule B), we can probably remove this check since the last node should always be there. What do you think, @rniemeyer?